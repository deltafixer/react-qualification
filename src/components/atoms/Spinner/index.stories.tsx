import React from "react";
import Spinner from "./";

export default {
  component: Spinner,
  title: "Spinner",
};

// hardcoding styled div wrapper because storybooks theme is dark, app is white
// would make it smarter if it wasn't a test project (not to go deep in this when no need)

export const Default = () => (
  <div style={{ background: "white", width: "fit-content", margin: "auto" }}>
    <Spinner />
  </div>
);

export const WithText = () => (
  <div style={{ background: "white", width: "fit-content", margin: "auto" }}>
    <Spinner text="Loading" />
  </div>
);
