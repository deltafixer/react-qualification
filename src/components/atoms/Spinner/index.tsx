import React from "react";
import { SpinnerStyleWrapper } from "./index.styles";

type SpinnerProps = {
  text: string;
};

const Spinner = ({ text }: SpinnerProps): JSX.Element => {
  return (
    <SpinnerStyleWrapper>
      <div className="loader"></div>
      <div className="loading">{text}</div>
    </SpinnerStyleWrapper>
  );
};

Spinner.defaultProps = {
  text: "",
};

export default Spinner;
