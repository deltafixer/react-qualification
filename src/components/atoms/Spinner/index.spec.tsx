import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import Spinner from "./index";

let wrapper: ShallowWrapper;

describe("Spinner component tests", () => {
  beforeAll(() => {
    wrapper = shallow(<Spinner />);
  });

  it("renders correctly", () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });
});
