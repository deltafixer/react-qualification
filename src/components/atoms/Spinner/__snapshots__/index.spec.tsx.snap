// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner component tests renders correctly 1`] = `
"<styled.div>
  <div className=\\"loader\\" />
  <div className=\\"loading\\" />
</styled.div>"
`;
